test/denormalized_has_one_test.rb

```diff
@@ -120,4 +120,45 @@ def test_cache_has_one_on_derived_model_raises
       StiRecordTypeA.cache_has_one :polymorphic_record, :inverse_name => :owner, :embed => true
     end
   end
+
+  def test_returned_record_should_be_readonly_on_cache_hit
+    IdentityCache.with_fetch_read_only_records do
+      Item.fetch_by_title('foo')
```

Comment:
Using `Item.fetch(1)` on this line and the following line would make sense though, since the focus on this test isn't on `fetch_by_title`, so we may as well skip that layer of caching